New data source: `azurerm_monitor_workspace` (#23928)
hashicorp/terraform-provider-azurerm: website/docs/d/monitor_workspace.html.markdown

---
subcategory: "Monitor"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_monitor_data_collection_endpoint"
description: |-
  Get information about the specified Workspace.

---

# Data Source: azurerm_monitor_workspace

Use this data source to access information about an existing Workspace.

## Example Usage

```hcl
data "azurerm_monitor_workspace" "example" {
  name                = "example-workspace"
  resource_group_name = azurerm_resource_group.example.name
}

output "query_endpoint" {
  value = data.azurerm_monitor_workspace.example.query_endpoint
}
```

## Argument Reference

* `name` - Specifies the name of the Workspace.

* `resource_group_name` - Specifies the name of the resource group the Workspace is located in.

## Attributes Reference

* `id` - The ID of the Resource.

* `kind` - The kind of the Workspace. Possible values are `Linux` and `Windows`.

* `location` - The Azure Region where the Workspace is located.

* `query_endpoint` - The query endpoint for the Azure Monitor Workspace.

> * `public_network_access_enabled` - Whether network access from public internet to the Workspace are allowed. Possible values are `true` and `false`.

Reviewer: ```suggestion
* `public_network_access_enabled` - Whether network access from public internet to the Workspace are allowed.
```

> * `tags` - A mapping of tags which should be assigned to the Data Collection Endpoint.

Reviewer: ```suggestion
* `tags` - A mapping of tags that are assigned to the Workspace.
```
